### ISSUE TITLE
Issue with refinement on structure member with a model v1

### ISSUE THREAD

**Issue by daddykotex** [disneystreaming/smithy4s#791]
When parsing a V1 model, Smithy will add `DefaultTrait` under some conditions. One of the refinements examples we have in this repo is affected by this:
1. a root shape is introduced: `integer Age`
2. it is used in V1 model

The member shape gets a `DefaultTrait` added to it, and so when we render the refinement type, we don't wrap in `Option`. But we also don't set a default value.

Documented here: https://smithy.io/2.0/guides/migrating-idl-1-to-2.html?highlight=boxed#converting-primitive-root-level-shapes-from-1-0

Source is [in smithy-model](https://github.com/awslabs/smithy/blob/56263d64885688118225acfba565d5eea6d1bc3c/smithy-model/src/main/java/software/amazon/smithy/model/loader/ModelInteropTransformer.java)

What should we do?

-------

I've setup a small scala-cli script just to confirm that this is what is actually going on:

```scala
//> using scala "2.13.10"
//> using lib "software.amazon.smithy:smithy-model:1.27.0"

import software.amazon.smithy.model.Model
import scala.jdk.CollectionConverters._
import software.amazon.smithy.model.shapes.ShapeId

object Main extends App {
  val model = Model
    .assembler()
    .addUnparsedModel(
      "defaults.smithy",
      s"""|namespace atest
          |
          |structure DirectInt {
          |  member: Integer
          |}
          |
          |integer MyInt
          |structure IndirectInt {
          |  member: MyInt
          |}
          |""".stripMargin
    )
    .assemble()
    .unwrap()
  println(
    model.expectShape(ShapeId.from("atest#DirectInt$member")).getAllTraits()
  )
  println(
    model.expectShape(ShapeId.from("atest#IndirectInt$member")).getAllTraits()
  )
// {}
// {smithy.api#default=Trait `smithy.api#default`, defined at defaults.smithy [9, 3]}
}
```

**Comment by lewisjkl** [comment]
Kind of related (inverse?) discussion here: https://github.com/disneystreaming/smithy4s/pull/787#discussion_r1097633603

**Comment by daddykotex** [comment]
yes, I used your example to generate mine :D

**Comment by Baccata** [comment]
> What should we do?

I believe we should wrap it in Option, in this case. Most of the time, it hardly makes sense for a refined type to have a default value.

**Comment by daddykotex** [comment]
To be clear:  we found the issue because the model v1 loading adds a default trait.

But the issue is more general. If I update the `refined.smithy` file in example module with the following diff:

```diff
diff --git a/sampleSpecs/refined.smithy b/sampleSpecs/refined.smithy
index eed036dc..52ab5143 100644
--- a/sampleSpecs/refined.smithy
+++ b/sampleSpecs/refined.smithy
@@ -1,3 +1,5 @@
+$version: "2"
+
 namespace smithy4s.example
 
 use smithy4s.meta#refinement
@@ -88,6 +90,8 @@ string DogName
 structure StructureWithRefinedTypes {
   age: Age,
   personAge: PersonAge,
+  @default(1)
+  otherAge: PersonAge,
   fancyList: FancyList,
   unwrappedFancyList: UnwrappedFancyList,
   name: Name,
```

I add a member with a default. I also bump the idl language (unfortunate, but I have to) to v2.
This produce the following diff:

```diff
diff --git a/modules/example/src/smithy4s/example/StructureWithRefinedTypes.scala b/modules/example/src/smithy4s/example/StructureWithRefinedTypes.scala
index 4ba74357..21fda0e9 100644
--- a/modules/example/src/smithy4s/example/StructureWithRefinedTypes.scala
+++ b/modules/example/src/smithy4s/example/StructureWithRefinedTypes.scala
@@ -6,15 +6,16 @@ import smithy4s.ShapeId
 import smithy4s.ShapeTag
 import smithy4s.schema.Schema.struct
 
-case class StructureWithRefinedTypes(age: Age, personAge: PersonAge, fancyList: Option[smithy4s.example.FancyList] = None, unwrappedFancyList: Option[smithy4s.example.refined.FancyList] = None, name: Option[smithy4s.example.Name] = None, dogName: Option[smithy4s.example.refined.Name] = None)
+case class StructureWithRefinedTypes(otherAge: PersonAge, age: Option[Age] = None, personAge: Option[PersonAge] = None, fancyList: Option[smithy4s.example.FancyList] = None, unwrappedFancyList: Option[smithy4s.example.refined.FancyList] = None, name: Option[smithy4s.example.Name] = None, dogName: Option[smithy4s.example.refined.Name] = None)
 object StructureWithRefinedTypes extends ShapeTag.Companion[StructureWithRefinedTypes] {
   val id: ShapeId = ShapeId("smithy4s.example", "StructureWithRefinedTypes")
 
   val hints: Hints = Hints.empty
 
   implicit val schema: Schema[StructureWithRefinedTypes] = struct(
-    Age.schema.required[StructureWithRefinedTypes]("age", _.age).addHints(smithy.api.Default(smithy4s.Document.fromDouble(0.0d))),
-    PersonAge.schema.required[StructureWithRefinedTypes]("personAge", _.personAge).addHints(smithy.api.Default(smithy4s.Document.fromDouble(0.0d))),
+    PersonAge.schema.required[StructureWithRefinedTypes]("otherAge", _.otherAge).addHints(smithy.api.Default(smithy4s.Document.fromDouble(1.0d))),
+    Age.schema.optional[StructureWithRefinedTypes]("age", _.age),
+    PersonAge.schema.optional[StructureWithRefinedTypes]("personAge", _.personAge),
     smithy4s.example.FancyList.schema.optional[StructureWithRefinedTypes]("fancyList", _.fancyList),
     UnwrappedFancyList.underlyingSchema.optional[StructureWithRefinedTypes]("unwrappedFancyList", _.unwrappedFancyList),
     smithy4s.example.Name.schema.optional[StructureWithRefinedTypes]("name", _.name),

```

This expose two things:
1. we see the issue because now the default trait that's implicitly added when using v1 is gone. the result is that `age` and `personAge` get wrapped into `Option`
2. the `otherAge` is not wrapped into an `Option` because we've defined a `default` for it. **but** *the default is ignored*, ie: we don't see `otherAge: PersonAge = PersonAge(1)` because it would not make sense to do that

So in general, the issue is that `default` are ignored when rendering the code related to a refined type. It just happen that we noticed it because of implicit default being added in the v1 mode.

So my fix in #795 is to say, fine we ignore the default, in that case, leave the field in a `Option`.

**Comment by Baccata** [comment]
I think this can be closed
